.travis.yml

```diff
@@ -127,82 +139,108 @@ linux_with_fuse: &linux_with_fuse
     # Increase the max number of user watches to ensure that watchman is able to watch all
     # files in the working copy.
     - sudo sysctl fs.inotify.max_user_watches=524288
-    - sudo pip install awscli
-    # Multipart operations aren't supported for anonymous users, so we set the
-    # threshold high to avoid them being used automatically by the aws cli.
-    - aws configure set default.s3.multipart_threshold 1024MB
-    - sudo apt-get install -y jq unzip
+    - ./build-support/bin/install_aws_cli_for_ci.sh
     - sudo apt-get install -y pkg-config fuse libfuse-dev
     - sudo modprobe fuse
     - sudo chmod 666 /dev/fuse
     - sudo chown root:$USER /etc/fuse.conf
 
 matrix:
   include:
-    - <<: *default_test_config
-      name: "Bootstrap Pants"
+    - <<: *default_linux_config
+      <<: *native_engine_cache_config
+      name: "Build Linux native engine and pants.pex"
       stage: *bootstrap
+      services:
+        - docker
+      env:
+        - CACHE_NAME=linuxpexbuild
```

Comment:
The python interpreter was recently unsafe to cache because the call site makes a symlink. Practically fine if you use the system /usr/bin/python or something, but breaks for us since we bootstrap our interpreter from a virtualenv.

I haven't looked to see if that behavior has changed in the last couple months, but I ended up moving the interpreter cache under .pants.d. That allows me to continue caching all of XDG_CACHE_HOME/pants.

Since this is for Pants CI, caching is likely fine, but FYI